hydromt_wflow/workflows/demand.py

```diff
@@ -0,0 +1,839 @@
+    # Set nodata values to zeros
+    waterareas_mr = waterareas_mr.where(
+        waterareas_mr != waterareas_mr.raster.nodata,
+        0,
+    )
+
+    # Get the fractions based on area count
+    w_pixels = np.take(
+        np.bincount(
+            waterareas_mr.values[x, y],
+            weights=gwbodies_mr.values[x, y],
+        ),
+        waterareas_mr.values[x, y],
+    )
+    a_pixels = np.take(
+        np.bincount(
+            waterareas_mr.values[x, y],
+            weights=gwbodies_mr.values[x, y] * 0.0 + 1.0,
+        ),
+        waterareas_mr.values[x, y],
+    )
+
+    # Determine the groundwater fraction
+    gwfrac_val = np.minimum(
+        gwfrac_raw_mr.values[x, y] * (a_pixels / (w_pixels + 0.01)),
+        1 - ncfrac_mr.values[x, y],
+    )
+    gwfrac_val[np.where(gwbodies_mr.values[x, y] == 0)] = 0
+    # invert to get surface water frac
+    gwfrac_val = 1 - gwfrac_val
+
+    # create the dataarray for the fraction
+    swfrac = xr.full_like(
+        gwfrac_raw_mr,
+        fill_value=np.nan,
+        dtype=np.float32,
+    ).load()
+    swfrac.name = "SurfaceWaterFrac"
+    swfrac.attrs = {"_FillValue": -9999}
+    swfrac = swfrac.copy()
+
+    # Set and interpolate the values
+    swfrac.values[x, y] = gwfrac_val
+    if interpolate:
+        swfrac = swfrac.interpolate_na(dim=swfrac.raster.x_dim, method="linear")
+        swfrac = swfrac.interpolate_na(
+            dim=swfrac.raster.x_dim, method="linear", fill_value="extrapolate"
+        )
+
+    # Set the nodata values based on the dem of the model (da_like)
+    swfrac = swfrac.where(da_like != da_like.raster.nodata, -9999)
+
+    # Return surface water frac
+    return swfrac
+
+
+def classify_pixels(
+    da_crop: xr.DataArray,
+    da_model: xr.DataArray,
+    threshold: float,
+    nodata_value: float | int = -9999,
+):
+    """Classifies pixels based on a (fractional) threshold.
+
+    Pixels with a value above this threshold are set to be irrigated, and pixels below
+    this value as rainfed.
+
+    Parameters
+    ----------
+    da_crop: xr.DataArray
+        Data with crop mask
+    da_model: xr.DataArray
+        Layer of the wflow model (used for reprojecting)
+    threshold: float
+        Threshold above which pixels are classified as irrigated
+    nodata_value: float | int = -9999
+        Value to be used for nodata
+
+    Returns
+    -------
+    crop_map: xr.DataArray
+        Mask with classified pixels
+    """
+    # Convert crop map to an area map
+    da_area = da_crop * da_crop.raster.area_grid()
+    # Resample to model grid and sum areas
+    da_area2model = da_area.raster.reproject_like(da_model, method="sum")
+
+    # Calculate relative area
+    relative_area = da_area2model / da_area2model.raster.area_grid()
+
+    # Classify pixels with a 1 where it exceeds the threshold, and 0 where it doesnt.
+    crop_map = xr.where(relative_area >= threshold, 1, 0)
+    crop_map = xr.where(da_model.isnull(), nodata_value, crop_map)
```

Comment:
```suggestion
    crop_map = crop_map.where(da_model != da_model.raster.nodata, nodata_value)
```
To avoid using raster.mask_nodata in wflow.py. Also xr.where looses all attributes so it's usually better to use da.where for this kind of operations :)